build-tools/packages/build-tools/src/repoPolicyCheck/handlers/assertShortCode.ts

```diff
@@ -14,42 +23,35 @@ const codeToMsgMap = new Map<string, string>();
 let maxShortCode = -1;
 
 function getCallsiteString(msg: Node) {
-	return `${msg.getSourceFile().getFilePath()}@${msg.getStartLineNumber()}`;
+	// Use filepath:line number so that the error message can be navigated to by clicking on it in vscode.
+	return `${msg.getSourceFile().getFilePath()}:${msg.getStartLineNumber()}`;
 }
 
 /**
- * Given a source file this function will look for all assert functions contained in it, and return the second parameter from
+ * Map from assertion function name to the index of its message argument.
+ */
+const assertionFunctions: ReadonlyMap<string, number> = new Map([
```

Comment:
IMHO the ideal place here is similar to what we have in devmain (Office main repo) re tagging.
The way we solve these problems there are the following:
- there is shared config file that describes function to tag
- each project (in our case - package) can define its own functions to tag.

Canonical example of the need for second bucket - getStringInstance, assertBoolInstance, assertNodeCoreInstance in ODSP driver.

It would be great to start on this path, i.e. get this list out of code into config file.

Reply:
It would be also great to add unreachableCase() to the mix.

Oh, one thing to add - we do not conflict names and types in C++ code.
I.e. assert(bool, string) -> assertTag(bool, number).

Not sure if it matters for our case (JS), but we can cut 2 bytes from code (fail('0x5') vs. fail(0x5)) if we use numbers in the code (but format them into strings in telemetry / error objects).

Reply:
I think you are right that this list should not be hard coded.
TO avoid delaying the quality work, I merged the general improvements from this PR in #14246 .
I may followup later to try and setup a config based approach.

Currently we do not do not run it on package scope, and require all the short codes to go into a single central file. With careful setup I think this can be kept working while enabling per project configuration and making use outside our repo possible (though such users would have to use different assert functions to avoid colliding with our codes).

Reply:
devmain (that I'll keep as an example) also uses central DB for tags (it's not just across packages, it's shared for all released versions to ensure tags are not duplicated, to make it easier for developers to deal with them). But what to tag is project specific, with some "base" being the same for all projects.
I'd love us to get to somewhat similar model. Depending on the exactly goals we setup here, tagged DB might need to be external (especially if synchronization is done across releases)